src/query/service/src/pipelines/processors/transforms/transform_udf_script.rs

```diff
@@ -78,62 +238,178 @@ impl Transform for TransformUdfScript {
     const NAME: &'static str = "UDFScriptTransform";
 
     fn transform(&mut self, mut data_block: DataBlock) -> Result<DataBlock> {
-        let num_rows = data_block.num_rows();
         for func in &self.funcs {
-            // construct input record_batch
-            let block_entries = func
-                .arg_indices
-                .iter()
-                .map(|i| {
-                    let arg = data_block.get_by_offset(*i).clone();
-                    if contains_variant(&arg.data_type) {
-                        let new_arg = BlockEntry::new(
-                            arg.data_type.clone(),
-                            transform_variant(&arg.value, true)?,
-                        );
-                        Ok(new_arg)
-                    } else {
-                        Ok(arg)
+            let num_rows = data_block.num_rows();
+            let block_entries = self.prepare_block_entries(&func, &data_block)?;
+            let input_batch = self.create_input_batch(block_entries, num_rows)?;
+            let runtime_key = Self::get_runtime_key(&func)?;
+
+            if let Some(runtime) = self.script_runtimes.get(&runtime_key) {
+                let result_batch = runtime.handle_execution(&func, &input_batch)?;
+                self.update_datablock(func, result_batch, &mut data_block)?;
+            } else {
+                return Err(ErrorCode::UDFDataError(format!(
+                    "Failed to find runtime for function '{}' with key: {}",
+                    func.name, runtime_key
+                )));
+            }
+        }
+        Ok(data_block)
+    }
+}
+
+impl TransformUdfScript {
+    fn get_runtime_key(func: &UdfFunctionDesc) -> Result<String, ErrorCode> {
+        if let Some((lang, _, code)) = func.udf_type.as_script() {
```

Comment:
I think we can use `func.name` as key, it's ensured to be unique in create udf statement.

Reply:
I'm concerned that using `func.name` as a key could potentially lead to a higher number of script runtime instances. I may need some additional time to research this further and will get back to you.

Reply:
- Enhanced **get_runtime_key():** key generation, now it consider both language (`lang`) and function name (`func.func_name`).